client/src/leap/soledad/client/api.py

```diff
@@ -70,6 +71,26 @@
 """
 SOLEDAD_CERT = None
 
+# A whitelist of modules from where to collect plugins dynamically.
+# For the moment restricted to leap namespace, but the idea is that we can pass
+# other "trusted" modules as options to the initialization of soledad.
+PLUGGABLE_LEAP_MODULES = ('mail', 'keymanager')
+
+
+def collect_plugins(interface):
+    """
+    Traverse a whitelist of modules and collect all the plugins that implement
+    the passed interface.
+    """
+    plugins = []
+    for namespace in PLUGGABLE_LEAP_MODULES:
+        try:
+            module = __import__('leap.%s.plugins' % namespace, fromlist='.')
+            plugins = plugins + list(getPlugins(interface, module))
```

Comment:
>  this depends on the idea of the whitelist (we don't want to traverse all imported modules, i guess)

see my recent comment above. a hardcoded whitelist as default + a configured list / wildcard that can be modified by a config doesn't seem so wild to mee.

> and would be a behaviour that depends on a certain folder structure on the modules.

in the twisted ecosystem is pretty standard to include a twisted/plugins folder.
for many apps I think it's pretty standard to drop folders in a plugins/ folder too. 

> I would prefer to not enforce a certain folder structure and adding the registration function

I see your point, and I think that might work, although I have my fears that w/o any means of control this could potentially affect the stability of the application. God forgive me if this sounds like a defense of Control with Capital C, but I think it's one of the biggest problems of opening the door to plugins.